scripts/dev/tidy.php

```diff
@@ -0,0 +1,394 @@
+#!/usr/bin/env php
+<?php
+
+/*
+  +----------------------------------------------------------------------+
+  | PHP Version 7                                                        |
+  +----------------------------------------------------------------------+
+  | Copyright (c) 1997-2018 The PHP Group                                |
+  +----------------------------------------------------------------------+
+  | This source file is subject to version 3.01 of the PHP license,      |
+  | that is bundled with this package in the file LICENSE, and is        |
+  | available through the world-wide-web at the following url:           |
+  | https://php.net/license/3_01.txt                                     |
+  | If you did not receive a copy of the PHP license and are unable to   |
+  | obtain it through the world-wide-web, please send a note to          |
+  | license@php.net so we can mail you a copy immediately.               |
+  +----------------------------------------------------------------------+
+*/
+
+/**
+ * Script usage and help information.
+ */
+function usage(): string
+{
+    return <<<END
+  PHP tidy script
+
+  SYNOPSIS:
+
+    php tidy.php [OPTION...] input-path
+
+  DESCRIPTION:
+
+    Trims trailing whitespace (spaces and tabs) from each line and trims
+    multiple final new lines into a single new line. If file has different
+    line endings the prevailing EOL (end of line) character is used as a
+    final new line.
+
+    When used on the php-src root directory it ignores the predefined set
+    of binary files and certain test files.
+
+    By default, original files are backed up (file.ext~).
+
+  OPTIONS:
+
+    -h, --help           Display this help
+    -v, --version        Display version
+        --no-whitespace  Disable trimming trailing whitespace
+        --no-eol         Disable fixing final end of lines
+        --no-colors      Disable colors in the script output
+        --no-backup      Disable backed up files. Use with caution to not
+                         overwrite something by accident.
+    -f, --fix            Edit and save files in place
+    -q, --quiet          Do not output any message
+
+  EXAMPLES:
+
+    Display help: php tidy.php -h
+
+    Check php-src: scripts/dev/tidy.php .
+
+    Fix php-src: scripts/dev/tidy.php --fix --no-backup .
+
+    Fix custom path or file: ./tidy.php --fix input-path
+
+END;
+}
+
+/**
+ * Determine the script version. If script is located in the PHP repository the
+ * PHP version is returned, otherwise the script's file sha1 hash is returned.
+ */
+function version(): string
```

Comment:
This does not seem useful?

Reply:
Yes, true. A bit over complicated. Removed.